components/Typography/tests/unit/MaterialScalableFontTests.m

```diff
@@ -0,0 +1,266 @@
+    UIContentSizeCategoryExtraSmall,
+    UIContentSizeCategorySmall,
+    UIContentSizeCategoryMedium,
+    UIContentSizeCategoryLarge,
+    UIContentSizeCategoryExtraLarge,
+    UIContentSizeCategoryExtraExtraLarge,
+    UIContentSizeCategoryExtraExtraExtraLarge,
+    UIContentSizeCategoryAccessibilityMedium,
+    UIContentSizeCategoryAccessibilityLarge,
+    UIContentSizeCategoryAccessibilityExtraLarge,
+    UIContentSizeCategoryAccessibilityExtraExtraLarge,
+    UIContentSizeCategoryAccessibilityExtraExtraExtraLarge,
+  ];
+
+  for (MDCTextStyle textStyle in textStyles) {
+    // When
+    UIFont *font = [UIFont systemFontOfSize:18.0];
+
+    MDCFontScaler *scaler = [[MDCFontScaler alloc] initForMaterialTextStyle:textStyle];
+
+    UIFont *scalableFont = [scaler scalableFontWithFont:font];
+
+    for (unsigned long ii = 0; ii < sizeCategories.count - 1; ++ii) {
+      UIContentSizeCategory smallerSizeCategory = sizeCategories[ii];
+      UIContentSizeCategory largerSizeCategory = sizeCategories[ii + 1];
+
+      UIFont *smallerFont = [scalableFont mdc_scaledFontForSizeCategory:smallerSizeCategory];
+      UIFont *largerFont = [scalableFont mdc_scaledFontForSizeCategory:largerSizeCategory];
+
+      // Then
+      XCTAssert(smallerFont.pointSize <= largerFont.pointSize);
+    }
+  }
+}
+
+- (void)testScaledFontDefaultEqualsLarge {
+  // Given
+  NSArray<MDCTextStyle> *textStyles = @[
+    MDCTextStyleHeadline1,
+    MDCTextStyleHeadline2,
+    MDCTextStyleHeadline3,
+    MDCTextStyleHeadline4,
+    MDCTextStyleHeadline5,
+    MDCTextStyleHeadline6,
+    MDCTextStyleSubtitle1,
+    MDCTextStyleSubtitle2,
+    MDCTextStyleBody1,
+    MDCTextStyleBody2,
+    MDCTextStyleButton,
+    MDCTextStyleCaption,
+    MDCTextStyleOverline,
+  ];
+
+  for (MDCTextStyle textStyle in textStyles) {
+    // When
+    UIFont *font = [UIFont systemFontOfSize:18.0];
+
+    MDCFontScaler *scaler = [[MDCFontScaler alloc] initForMaterialTextStyle:textStyle];
+
+    UIFont *scalabledFont = [scaler scalableFontWithFont:font];
+
+    UIFont *defaultFont = [scalabledFont mdc_scaledFontAtDefaultSize];
+    UIFont *largeFont = [scalabledFont mdc_scaledFontForSizeCategory:UIContentSizeCategoryLarge];
+
+    // Then
+    XCTAssert([defaultFont mdc_isSimplyEqual:largeFont]);
+  }
+}
+
+- (void)testInvalidStyleFallback {
+  // Given
+  UIFont *font = [UIFont systemFontOfSize:18.0];
+
+  MDCFontScaler *scaler1 = [[MDCFontScaler alloc] initForMaterialTextStyle:MDCTextStyleBody1];
+  MDCFontScaler *scaler2 =
+      [[MDCFontScaler alloc] initForMaterialTextStyle:@"IntentionallyNonTextStyleString"];
+
+  UIFont *scalableFont1 = [scaler1 scalableFontWithFont:font];
+  UIFont *scalableFont2 = [scaler2 scalableFontWithFont:font];
+
+  UIFont *scaledFont1 = [scalableFont1 mdc_scaledFontAtDefaultSize];
+  UIFont *scaledFont2 = [scalableFont2 mdc_scaledFontAtDefaultSize];
+
+  // Then
+  XCTAssert([scaledFont1 mdc_isSimplyEqual:scaledFont2]);
```

Comment:
I believe a better assertion is that `scaledFont2` is the same object as `font`.  I  believe  neither `scalableFont1` nor `scaledFont1` are needed.  If that's not the case, please consider renaming the method to make the expectation more clear.

What you seem to be testing here is that scaling a font to `body1` and then back to the default is equivalent to using an invalid text style.